src/vm/prestub.cpp

```diff
@@ -303,6 +303,12 @@ PCODE MethodDesc::PrepareILBasedCode(PrepareCodeConfig* pConfig)
     }
     if (pCode == NULL)
     {
+        if (pConfig->GetMethodDesc()->IsEligibleForTieredCompilation() &&
```

Comment:
I'd suggest moving this down to JitCompileCodeLockedEventWrapper to eliminate notifications that would otherwise be generated from multicore jit code. I would also change the conditional to be 

    g_pConfig->TieredCompilation() && !flags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_TIER1)

This ensures that any jitting other than tier1 will defer tier1. For example if there was a large amount of non-versionable code being jitted it would not qualify for your current check but it would qualify for the !tier1 check. This check is also probably a little faster to evaluate.

Reply:
Oops missed the non-versionable code again. I'll move it to that function, makes sense.

I was thinking profiler-triggered jitting shouldn't affect the delay since it doesn't change when app startup completes. Do you think it should? How about:
`g_pConfig->TieredCompilation() && pConfig->GetCodeVersion().IsDefaultVersion()`

Reply:
I'll admit its largely guesswork until we have a scenario where it matters, but I think a typical usage of profiler ReJIT is actually to replace the IL before the method has been called the first time. In that case the startup path is the non-default profiler instrumented version of the code.

Reply:
I see, will fix